data/src/main/java/com/antosik/benchproject/data/movie/details/repo/MovieDetailsDataRepository.kt

```diff
@@ -1,12 +1,25 @@
 package com.antosik.benchproject.data.movie.details.repo
 
+import com.antosik.benchproject.data.movie.details.entity.toEntity
 import com.antosik.benchproject.data.movies.common.api.MoviesApi
+import com.antosik.benchproject.data.movies.common.database.MovieDao
 import com.antosik.benchproject.domain.movie.details.repo.MovieDetailsRepository
 
 internal class MovieDetailsDataRepository(
-    private val apiService: MoviesApi
+    private val apiService: MoviesApi,
+    private val dao: MovieDao
 ) : MovieDetailsRepository {
     override suspend fun getMovieDetails(movieId: Int) = runCatching {
-        apiService.getMovieDetails(movieId).toDomain()
+        val movieDetailsRemote = apiService.getMovieDetails(movieId)
+        dao.insertMovieDetails(movieDetailsRemote.toEntity())
+        dao.insertGenres(movieDetailsRemote.genres.toEntity(movieId))
+    }.run {
+        val movieDetailsEntity = dao.getMovieAndMovieDetailsWithId(movieId).firstOrNull()?.movieDetailsEntity
+        val genresEntity = dao.getMovieDetailsWithGenres(movieId).firstOrNull()?.genresEntity
+        if (genresEntity != null) {
+            movieDetailsEntity?.toDomain(genresEntity)
```

Comment:
Couldn't we benefit from Room relations to get one object which holds both genres and movie details? To call get on Room only once and to avoid passing a parameter to the mapper. Maybe I don't see something here and it's not possible, but please double-check :D

Reply:
I think I found the way how to use benefits from the relation between tables